File: client/app/src/main/java/nl/hva/capstone/api/model/output/Conversation.kt
package nl.hva.capstone.api.model.output

import androidx.compose.runtime.Immutable
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@Immutable
data class Conversation(
  val conversation: RawConversation,
  @SerialName("other_participant") val otherParticipant: User,
  @SerialName("last_message") val lastMessage: ConversationMessage?,
) {
  val id get() = conversation.id
}

File: backend/src/ws/conversation.rs
use serde::Serialize;
use typeshare::typeshare;

use crate::db::entities::*;

#[derive(Serialize, Debug)]
#[typeshare]
pub struct FullConversation {
  pub conversation: conversation::Model,
  pub other_participant: user::Model,
  pub last_message: Option<conversation_message::Model>,
}
